Framework/Algorithms/test/ExtractMaskTest.h

```diff
@@ -91,29 +91,27 @@ class ExtractMaskTest : public CxxTest::TestSuite {
     const auto &oSpecInfo = outputWS->spectrumInfo();
     for (size_t i = 0; i < nOutputHists; ++i) {
       // Sizes
-      TS_ASSERT_EQUALS(outputWS->readX(i).size(), 1);
-      TS_ASSERT_EQUALS(outputWS->readY(i).size(), 1);
-      TS_ASSERT_EQUALS(outputWS->readE(i).size(), 1);
+      TS_ASSERT_EQUALS(outputWS->x(i).size(), 1);
+      TS_ASSERT_EQUALS(outputWS->y(i).size(), 1);
+      TS_ASSERT_EQUALS(outputWS->e(i).size(), 1);
       // Data
       double expectedValue(-1.0);
-      bool outputMasked(false);
       if (!iSpecInfo.hasDetectors(i) || !oSpecInfo.hasDetectors(i)) {
         expectedValue = 1.0;
       }
 
       if (iSpecInfo.hasDetectors(i) && iSpecInfo.isMasked(i)) {
         expectedValue = 1.0;
-        outputMasked = true;
       } else {
         expectedValue = 0.0;
-        outputMasked = false;
       }
 
-      TS_ASSERT_EQUALS(outputWS->dataY(i)[0], expectedValue);
-      TS_ASSERT_EQUALS(outputWS->dataE(i)[0], expectedValue);
-      TS_ASSERT_EQUALS(outputWS->dataX(i)[0], 0.0);
-      if (iSpecInfo.hasDetectors(i)) {
-        TS_ASSERT_EQUALS(oSpecInfo.isMasked(i), outputMasked);
+      TS_ASSERT_EQUALS(outputWS->y(i)[0], expectedValue);
+      TS_ASSERT_EQUALS(outputWS->e(i)[0], 0.0);
+      TS_ASSERT_EQUALS(outputWS->x(i)[0], 1.0);
+      // Detectors never masked since masking information is carried by Y.
+      if (oSpecInfo.hasDetectors(i)) {
```

Comment:
The test had been disabled a long time ago because it was broken. The changes were necessary to fix it. The comment refers to the change in the assert one line below. `oSpecInfo` is used to be consistent with what is used to check masking in the line below, but probably does not matter since the instrument and spectrum mapping is the same in input and output,